src/lib/serviceRoutines/postQueryContext.cpp

```diff
@@ -241,26 +178,198 @@ std::string postQueryContext
   {
-    qcrsP->errorCode.fill(SccOk, std::string("Redirected to context provider ") + ip + ":" + portV + prefix);
+    qcrsP->errorCode.fill(SccContextElementNotFound);
   }
+}
+
+
 
-  if (qcrsP->contextElementResponseVector.size() > 0)
+/* ****************************************************************************
+*
+* postQueryContext -
+*/
+std::string postQueryContext
+(
+  ConnectionInfo*            ciP,
+  int                        components,
+  std::vector<std::string>&  compV,
+  ParseData*                 parseDataP
+)
+{
+  //
+  // Convops calling this routine may need the response in digital
+  // So, the digital response is passed back in parseDataP->qcrs.res
+  //
+  QueryContextResponse*       qcrsP = &parseDataP->qcrs.res;
+  std::string                 answer;
+  QueryContextRequestVector   requestV;
+  QueryContextResponseVector  responseV;
+
+  qcrsP->errorCode.fill(SccOk);
+  ciP->httpStatusCode = mongoQueryContext(&parseDataP->qcr.res, qcrsP, ciP->tenant, ciP->servicePathV, ciP->uriParam);
+
+  //
+  // In this loop the output from mongoQueryContext is examines and the attributes are sorted
+  // by their providing application, in requestV, later to be used to forward the queries to their
+  // respective Context Providers.
+  //
+  // The local part of the query is already tken care of by mongoQueryContext and the result must be moved
+  // to the response vector.
+  // All the local response will be gathered in one single instance of QueryContextResponse.
+  // As a "QueryContextResponse::ContextElementResponse::ContextElement" can contain only ONE entity,
+  // we will need a separate ContextElementResponse for each distinct EntityId in the local response.
+  // All attributes that belong to the same EntityId will be gathered in one unique instance of
+  // ContextElementResponse - so, the correct ContextElementResponse must be looked up and if not found,
+  // it must be created and added to the QueryContextResponse of the local response
+  //
+  QueryContextResponse* localQcrsP = new QueryContextResponse();
+
+  for (unsigned int ix = 0 ; ix < qcrsP->contextElementResponseVector.size(); ++ix)
   {
-    qcrsP->contextElementResponseVector[0]->statusCode.details = std::string("Redirected to context provider ") + ip + ":" + portV + prefix;
+    ContextElementResponse* cerP  = qcrsP->contextElementResponseVector[ix]->clone();
+
+    for (unsigned int aIx = 0 ; aIx < cerP->contextElement.contextAttributeVector.size(); ++aIx)
+    {
+      EntityId*            eP  = &cerP->contextElement.entityId;
+      ContextAttribute*    aP  = cerP->contextElement.contextAttributeVector[aIx];
+      
+      //
+      // An empty providingApplication means the attribute is local
+      // In such a case, the response is already in our hand, we just need to copy it to responseV
+      //
+      if (aP->providingApplication == "")
```

Comment:
Take into account that in this case maybe it more than just another test... As I mention above, I don't see any point in the code in which you process the ContextElement providingApplicationList field (maybe the code is there, but I haven's seen it).